service/src/test/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateSentinelStepTest.java

```diff
@@ -0,0 +1,124 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.landingzone.stairway.common.model.TargetManagedResourceGroup;
+import bio.terra.landingzone.stairway.flight.LandingZoneFlightMapKeys;
+import bio.terra.landingzone.stairway.flight.exception.MissingRequiredFieldsException;
+import bio.terra.profile.model.ProfileModel;
+import bio.terra.stairway.StepResult;
+import com.azure.resourcemanager.securityinsights.SecurityInsightsManager;
+import com.azure.resourcemanager.securityinsights.models.SentinelOnboardingState;
+import com.azure.resourcemanager.securityinsights.models.SentinelOnboardingStates;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag("unit")
+class CreateSentinelStepTest extends BaseStepTest {
+  private static final String RESOURCE_ID = "sentinelId";
+
+  @Mock SecurityInsightsManager mockSecurityInsightsManager;
+  @Mock SentinelOnboardingStates mockSentinelOnboardingStages;
+  @Mock SentinelOnboardingState.DefinitionStages.Blank mockDefinitionStagesBlank;
+  @Mock SentinelOnboardingState.DefinitionStages.WithCreate mockDefinitionStagesWithCreate;
+  @Mock SentinelOnboardingState mockSentinelOnboardingState;
+
+  private CreateSentinelStep createSentinelStep;
+
+  @BeforeEach
+  void setUp() {
+    createSentinelStep =
+        new CreateSentinelStep(mockArmManagers, mockParametersResolver, mockResourceNameGenerator);
+  }
+
+  @Test
+  void doStepSuccess() throws InterruptedException {
+    setupFlightContext(
+        mockFlightContext,
+        Map.of(
+            LandingZoneFlightMapKeys.BILLING_PROFILE,
+            new ProfileModel().id(UUID.randomUUID()),
+            LandingZoneFlightMapKeys.LANDING_ZONE_ID,
+            LANDING_ZONE_ID),
+        Map.of(
+            GetManagedResourceGroupInfo.TARGET_MRG_KEY,
+            new TargetManagedResourceGroup("mgrName", "mrgRegion"),
+            CreateLogAnalyticsWorkspaceStep.LOG_ANALYTICS_RESOURCE_KEY,
+            buildLandingZoneResource()));
+    setupArmManagersForDoStep(RESOURCE_ID);
+
+    StepResult stepResult = createSentinelStep.doStep(mockFlightContext);
+
+    assertThat(stepResult, equalTo(StepResult.getStepResultSuccess()));
```

Comment:
Are there assertions to be made against the security insights manger mock?

Reply:
I think important thing is to check in this "fluent API" flow if we are sending request to Azure to "onboard" sentinel. This is what is happening on the next line. I think now there is nothing to validate against insights manager mock.

Reply:
Add capturing for some arguments.